regions/shapes/circle.py

```diff
@@ -195,3 +195,15 @@ def to_pixel(self, wcs):
         center, scale, _ = skycoord_to_pixel_scale_angle(self.center, wcs)
         radius = self.radius.to('deg').value * scale
         return CirclePixelRegion(center, radius, self.meta, self.visual)
+
+
+class CircleSphericalRegion(CircleSkyRegion):
+    """Spherical circle sky region.
+
+    TODO: document
```

Comment:
See my comment on #276 .  I'd be happy to help review and/or write.

Reply:
@gpdf - Thanks!
I'll add some docs in this PR, and then you can review.